drivers/SmartThings/zigbee-window-treatment/src/aqara/aqara_utils.lua

```diff
@@ -0,0 +1,212 @@
+local capabilities = require "st.capabilities"
+local clusters = require "st.zigbee.zcl.clusters"
+local cluster_base = require "st.zigbee.cluster_base"
+local data_types = require "st.zigbee.data_types"
+local utils = require "st.utils"
+
+local Basic = clusters.Basic
+local WindowCovering = clusters.WindowCovering
+local AnalogOutput = clusters.AnalogOutput
+
+local PRIVATE_CLUSTER_ID = 0xFCC0
+local PRIVATE_ATTRIBUTE_ID = 0x0009
+local MFG_CODE = 0x115F
+local PREF_ATTRIBUTE_ID = 0x0401
+local SHADE_STATE_ATTRIBUTE_ID = 0x0404
+
+local SHADE_STATE_STOP = 0
+local SHADE_STATE_OPEN = 1
+local SHADE_STATE_CLOSE = 2
+local INITIALIZED_STATE = "initializedState"
+
+local PREF_REVERSE_DEFAULT = "\x00\x02\x00\x00\x00\x00\x00"
+local PREF_REVERSE_OFF = "\x00\x02\x00\x00\x00\x00\x00"
+local PREF_REVERSE_ON = "\x00\x02\x00\x01\x00\x00\x00"
+
+local FINGERPRINTS = {
+  { mfr = "LUMI", model = "lumi.curtain", device_profile = "curtain" },
+  { mfr = "LUMI", model = "lumi.curtain.v1", device_profile = "curtain" },
+  { mfr = "LUMI", model = "lumi.curtain.aq2", device_profile = "roller-shade" }
+}
+
+local aqara_utils = {}
+
+local function read_private_attribute(device, cluster_id, attribute_id)
+  local message = cluster_base.read_attribute(device, data_types.ClusterId(cluster_id), attribute_id)
+  message.body.zcl_header.frame_ctrl:set_mfg_specific()
+  message.body.zcl_header.mfg_code = data_types.validate_or_build_type(MFG_CODE, data_types.Uint16, "mfg_code")
+  device:send(message)
+end
```

Comment:
we provide this with `cluster_base.read_manufacturer_specific_attribute` in `local cluster_base = require "st.zigbee.cluster_base"`

There are examples in other drivers.

Reply:
I'll edit and push.